packages/ovm/src/contracts/ExecutionManager.sol

```diff
@@ -183,6 +183,7 @@ contract ExecutionManager is FullStateManager {
         require(eoaAddress != ZERO_ADDRESS, "Failed to recover signature");
         // Require nonce to be correct
         require(_nonce == getOvmContractNonce(eoaAddress), "Incorrect nonce!");
+        executionContext.ovmTxOrigin = eoaAddress;
```

Comment:
Should this be un-set after the call? If not, I think calls (not txs) could potentially return different things depending on the tx that occurred before them.

Reply:
I think origin stays the same throughout the transaction regardless of calls. From the yellow paper ORIGIN is "the sender of original transaction"

But executionContext does persist in storage across transactions correct? If that's the case then I think you're right.  We'll need to unset origin either after the call is made or inside of `executeUnsignedEOACall`

Reply:
Yes, it is a storage var, so it should persist. Might be best to unset it where it's set instead of on every call.

Reply:
ah good call: [Reset `executionContext.ovmTxOrigin` to the zero address after EOA calls](https://github.com/ethereum-optimism/optimism-monorepo/pull/13/commits/05babcd325953148ae4a40ed7a4d0d585777d834)